src/gui/statusspeedbutton.cpp

```diff
@@ -142,23 +180,31 @@ void StatusSpeedButton::paintEvent(QPaintEvent* /*event*/)
         hPos += iconSize().width() + m_buttonMargin;
     }
 
-    painter.drawStaticText(hPos, 0, m_currentSpeedText);
-    hPos += m_currentSpeedText.textWidth();
+    painter.drawStaticText(hPos, 0, m_currentSpeed.text());
+    hPos += m_currentSpeed.text().textWidth();
     hPos += m_separatorWidth;
-    if (m_speedLimitText.textWidth() > 0) {
-        painter.drawStaticText(hPos, 0, m_speedLimitText);
-        hPos += m_speedLimitText.textWidth();
+    if (m_speedLimit.text().textWidth() > 0) {
+        painter.drawStaticText(hPos, 0, m_speedLimit.text());
+        hPos += m_speedLimit.text().textWidth();
         hPos += m_separatorWidth;
     }
-    if (m_totalPayloadText.textWidth() > 0) {
-        painter.drawStaticText(hPos, 0, m_totalPayloadText);
+    if (m_totalPayload.text().textWidth() > 0) {
+        painter.drawStaticText(hPos, 0, m_totalPayload.text());
     }
 }
 
-QStaticText StatusSpeedButton::emptyText()
+QChar StatusSpeedButton::widestDecimalDigit()
 {
-    QStaticText res;
-    res.setTextFormat(Qt::PlainText);
-    res.setTextOption({Qt::AlignRight});
-    return res;
+    const QString digits = QLocale().toString(1234567890);
```

Comment:
why QLocale here, does it do anything special?

Reply:
I need localised digits as would be produced by `QString::arg()`, but here I don't need the template string.

Reply:
I understand it now.
Not related to this PR, just my opinion, do you really expect to see a localized number string here? I mean I prefer plain number, commas or anything else for separator is redundant IMO.

Reply:
I'm weak in locale-related things. Just wanted to avoid any assumptions that might fail.